src/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs

```diff
@@ -120,15 +120,13 @@ private static Type GetPublicType(Type t)
             // we cannot use the expression tree in a context which has only execution
             // permissions.  We should endeavour to translate constants into 
             // new constants which have public types.
-            TypeInfo typeInfo = t.GetTypeInfo();
-            if (typeInfo.IsGenericType && typeInfo.GetGenericTypeDefinition().GetTypeInfo().ImplementedInterfaces.Contains(typeof(IGrouping<,>)))
+            if (t.IsGenericType && t.GetGenericTypeDefinition().GetInterfaces().Contains(typeof(IGrouping<,>)))
```

Comment:
Close examination shows that this test can never pass, and couldn't with the first version ever in the repository for that matter, due to something related to https://blogs.msdn.microsoft.com/haibo_luo/2006/02/17/type-fullname-returns-null-when/ meaning that the `Type` found which is *a* `IGrouping<,>` is not *the* `IGrouping<,>` and the `Contains` call never returns true.

This is less of an issue since `Lookup<,>.Grouping` isn't a nested type any more, but should still be fixed.